### PR TITLE
No issue enable kube api for ci

### PR DESCRIPTION
**Pull request by RazRegev**
For production environments, there is no need to generate
CRDs (and having all the dependencies they require) in the
production severs.
By default, if NO_CRD_GENERATION env provided, skip the
generation part and copy the existing resources.yaml file.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,6 +58,7 @@ TESTING_PUBLIC_CONTAINER_REGISTRIES := quay.io,registry.svc.ci.openshift.org
 PUBLIC_CONTAINER_REGISTRIES := $(or ${PUBLIC_CONTAINER_REGISTRIES},$(TESTING_PUBLIC_CONTAINER_REGISTRIES))
 PODMAN_PULL_FLAG := $(or ${PODMAN_PULL_FLAG},--pull always)
 ENABLE_KUBE_API := $(or ${ENABLE_KUBE_API},false)
+NO_CRD_GENERATION := $(or ${NO_CRD_GENERATION},)
```
Comment (filanov) [inline Makefile:61]: Maybe `GENERATE_CRD` instead?

File: hack/generate.sh [hunk 2]
```diff
@@ -81,13 +81,20 @@ function generate_manifests() {
...
-    goimports -w  ${controller_path}
+    if [ "${GENERATE_CRD:-true}" == "true" ]; then
+        echo "Generating CRDs"
+        controller-gen ${crd_options} rbac:roleName=assisted-service-manager-role paths="./..." output:rbac:dir=${controller_rbac_path} \
+        webhook paths="./..." output:crd:artifacts:config=${controller_crd_path}/bases
+        kustomize build ${controller_crd_path} > ${controller_crd_path}/resources.yaml
+        cp ${controller_crd_path}/resources.yaml ${BUILD_FOLDER}/resources.yaml
+        controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
+        goimports -w  ${controller_path}
+        exit 0
```
Comment (filanov) [inline hack/generate.sh:92]: why exit? just continue and copy do the copy no?

Comment (RazRegev) [reply hack/generate.sh:92]: just for logging purposes, will remove

File: hack/generate.sh [hunk 3]
```diff
@@ -81,10 +81,15 @@ function generate_manifests() {
...
+    if [ "${GENERATE_CRD:-true}" == "true" ]; then
+        echo "Generating CRDs"
+        controller-gen ${crd_options} rbac:roleName=assisted-service-manager-role paths="./..." output:rbac:dir=${controller_rbac_path} \
+        webhook paths="./..." output:crd:artifacts:config=${controller_crd_path}/bases
+        kustomize build ${controller_crd_path} > ${controller_crd_path}/resources.yaml
+        controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
+    fi
+
+    cp ${controller_crd_path}/resources.yaml ${BUILD_FOLDER}/resources.yaml
     goimports -w  ${controller_path}
```
Comment (filanov) [inline hack/generate.sh:93]: it's part of the generation no need to run it always

Comment (RazRegev) [reply hack/generate.sh:93]: ops my mistake

**Review by filanov — APPROVED** [review]

**Review by YuviGold — APPROVED** [review]
LGTM

File: Makefile [hunk 4]
```diff
@@ -221,7 +222,7 @@ deploy-service-requirements: | deploy-namespace deploy-inventory-service-file
 	$(MAKE) deploy-role deploy-resources
 
 deploy-resources: generate-manifests
-	python3 ./tools/deploy_crd.py $(ENABLE_KUBE_API_CMD) --apply-manifest $(APPLY_MANIFEST) --profile "$(PROFILE)" --target "$(TARGET)"
+	python3 ./tools/deploy_crd.py $(ENABLE_KUBE_API_CMD) --apply-manifest $(APPLY_MANIFEST) --profile "$(PROFILE)" --target "$(TARGET)" --namespace "$(NAMESPACE)"
```
Comment (filanov) [inline Makefile:225]: please break the line.

**Pull request merged**